esx_service/vmdk_ops.py

```diff
@@ -764,24 +773,45 @@ def executeRequest(vm_uuid, vm_name, config_path, cmd, full_vol_name, opts):
     Returns None (if all OK) or error string
     """
     logging.debug("config_path=%s", config_path)
-    vm_datastore = get_datastore_name(config_path)
+    vm_datastore_url = get_datastore_url_from_config_path(config_path)
+    vm_datastore = vmdk_utils.get_datastore_name(vm_datastore_url)
```

Comment:
this block spreads the knowledge about cache manipulation onto too many  places. 
I think it would be  way better to hide it , say in get_datastore_name() ... it can check get_datastores() and reset the cache internally if needed